driver-sync/src/test/functional/com/mongodb/client/csot/AbstractClientSideOperationsEncryptionTimeoutProseTest.java

```diff
@@ -0,0 +1,388 @@
+import com.mongodb.WriteConcern;
+import com.mongodb.client.Fixture;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.CreateCollectionOptions;
+import com.mongodb.client.model.CreateEncryptedCollectionParams;
+import com.mongodb.client.model.ValidationOptions;
+import com.mongodb.client.model.vault.DataKeyOptions;
+import com.mongodb.client.model.vault.EncryptOptions;
+import com.mongodb.client.test.CollectionHelper;
+import com.mongodb.client.vault.ClientEncryption;
+import com.mongodb.event.CommandStartedEvent;
+import com.mongodb.internal.connection.TestCommandListener;
+import org.bson.BsonBinary;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.bson.Document;
+import org.bson.codecs.BsonDocumentCodec;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.lessThan;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+/**
+ * See
+ * <a href="https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/tests/README.rst#clientencryption">Prose Tests</a>.
+ */
+public abstract class AbstractClientSideOperationsEncryptionTimeoutProseTest {
+
+    private static final Map<String, Map<String, Object>> KMS_PROVIDERS = new HashMap<>();
+
+    private final MongoNamespace keyVaultNamespace = new MongoNamespace("keyvault", "datakeys");
+
+    private CollectionHelper<BsonDocument> keyVaultCollectionHelper;
+
+    private TestCommandListener commandListener;
+
+    private static final String MASTER_KEY = "Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5a"
+            + "XRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk";
+
+    protected abstract ClientEncryption createClientEncryption(ClientEncryptionSettings.Builder builder);
+
+    protected abstract MongoClient createMongoClient(MongoClientSettings.Builder builder);
+
+    @Test
+    @Tag("setsFailPoint")
```

Comment:
I don't think this is a proper use of the `@Tag` functionality. We should either use something like `com.mongodb.client.FailPoint`, or implement automatic turning off of failpoints without using the `@Tag` functionality.

The original discussion is https://github.com/mongodb/mongo-java-driver/pull/1333#discussion_r1535047923.

Reply:
The `@Tag` annotation was removed as part of [this PR](https://github.com/mongodb/mongo-java-driver/pull/1383/files#r1611734757). Now, we ensure failpoints are disabled idempotently after each test in the `@AfterEach` method.